lib/facade_converter.ts

```diff
@@ -353,6 +353,50 @@ export class FacadeConverter extends base.TranspilerBase {
         this.emitMethodCall('remove', c.arguments);
         this.emit('!= null || true ) )');
       },
+      'Map.forEach': (c: ts.CallExpression, context: ts.Expression) => {
+        let cb: any;
+        let params: any;
+
+        switch (c.arguments[0].kind) {
+          case ts.SyntaxKind.FunctionExpression:
+            cb = <ts.FunctionExpression>(c.arguments[0]);
+            params = cb.parameters;
+            if (params.length != 2) {
+              this.reportError(c, 'Map.forEach callback requires exactly two arguments');
```

Comment:
Actually doesn't `reportError()` break the execution flow ? (otherwise there seems to be other places that need to be updated in this file)